session/src/main/java/org/apache/iotdb/session/SessionConnection.java

```diff
@@ -0,0 +1,401 @@
+    }
+  }
+
+
+  public void close() throws IoTDBConnectionException {
+    TSCloseSessionReq req = new TSCloseSessionReq(sessionId);
+    try {
+      client.closeSession(req);
+    } catch (TException e) {
+      throw new IoTDBConnectionException(
+          "Error occurs when closing session at server. Maybe server is down.", e);
+    } finally {
+      if (transport != null) {
+        transport.close();
+      }
+    }
+  }
+
+  protected void setTimeZone(String zoneId)
+      throws StatementExecutionException, IoTDBConnectionException {
+    TSSetTimeZoneReq req = new TSSetTimeZoneReq(sessionId, zoneId);
+    TSStatus resp;
+    try {
+      resp = client.setTimeZone(req);
+    } catch (TException e) {
+      throw new IoTDBConnectionException(e);
+    }
+    RpcUtils.verifySuccess(resp);
+    this.zoneId = ZoneId.of(zoneId);
+  }
+
+  protected String getTimeZone()
+      throws StatementExecutionException, IoTDBConnectionException {
+    if (zoneId != null) {
+      return zoneId.toString();
+    }
+    TSGetTimeZoneResp resp;
+    try {
+      resp = client.getTimeZone(sessionId);
+    } catch (TException e) {
+      throw new IoTDBConnectionException(e);
+    }
+    RpcUtils.verifySuccess(resp.getStatus());
+    return resp.getTimeZone();
+  }
+
+  protected void setStorageGroup(String storageGroup)
+      throws IoTDBConnectionException, StatementExecutionException, RedirectException {
+    try {
+      RpcUtils.verifySuccessWithRedirection(client.setStorageGroup(sessionId, storageGroup));
+    } catch (TException e) {
+      throw new IoTDBConnectionException(e);
+    }
+  }
+
+  protected void deleteStorageGroups(List<String> storageGroups)
+      throws IoTDBConnectionException, StatementExecutionException, RedirectException {
+    try {
+      RpcUtils.verifySuccessWithRedirection(client.deleteStorageGroups(sessionId, storageGroups));
+    } catch (TException e) {
+      throw new IoTDBConnectionException(e);
+    }
+  }
+
+  protected void createTimeseries(TSCreateTimeseriesReq request)
+      throws IoTDBConnectionException, StatementExecutionException {
+    request.setSessionId(sessionId);
+    try {
+      RpcUtils.verifySuccess(client.createTimeseries(request));
+    } catch (TException e) {
+      throw new IoTDBConnectionException(e);
+    }
+  }
+
+  protected void createMultiTimeseries(TSCreateMultiTimeseriesReq request)
+      throws IoTDBConnectionException, StatementExecutionException {
+    request.setSessionId(sessionId);
+    try {
+      RpcUtils.verifySuccess(client.createMultiTimeseries(request));
+    } catch (TException e) {
+      throw new IoTDBConnectionException(e);
+    }
+  }
+
+  protected boolean checkTimeseriesExists(String path)
+      throws IoTDBConnectionException, StatementExecutionException {
+    SessionDataSet dataSet = executeQueryStatement(String.format("SHOW TIMESERIES %s", path));
+    boolean result = dataSet.hasNext();
+    dataSet.closeOperationHandle();
```

Comment:
Close operation should be put into a finally block.

Reply:
OK~